catalog/spatial/geocoding/spatial-geocoding-extract/src/main/java/org/codice/ddf/spatial/geocoding/extract/GeoNamesFileExtractor.java

```diff
@@ -41,8 +44,14 @@
 import org.codice.ddf.spatial.geocoding.ProgressCallback;
 
 public class GeoNamesFileExtractor implements GeoEntryExtractor {
+
   private static final int BUFFER_SIZE = 4096;
 
+  private static final String[] SCHEMES = {"http", "https"};
+
+  private static final UrlValidator URL_VALIDATOR =
+      new UrlValidator(SCHEMES, UrlValidator.ALLOW_LOCAL_URLS);
```

Comment:
❓Is this UrlValidator really needed? Why not just try the resource as a file path, and , if that fails, try the resource as a URl, and, if that fails, try the resource as a keyword to automatically download a zip file from http://download.geonames.org/export/dump? You wouldn't need to start those extra three bundles.